pkg/controller/certificatesigningrequest/ipsec_csr_signing_controller.go

```diff
@@ -0,0 +1,473 @@
+		}
+		if !c.selfSignedCA {
+			klog.InfoS("Self-signed CA is disabled. Ensure CA Secret exists", "name", ipsecRootCAName, "namespace", env.GetAntreaNamespace())
+			return nil
+		}
+		caBytes, caKeyBytes, err = generateSelfSignedRootCertificate(ipsecRootCAName)
+		if err != nil {
+			return err
+		}
+		caSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      ipsecRootCAName,
+				Namespace: env.GetAntreaNamespace(),
+			},
+			Type: corev1.SecretTypeTLS,
+			Data: map[string][]byte{
+				corev1.TLSCertKey:       caBytes,
+				corev1.TLSPrivateKeyKey: caKeyBytes,
+			},
+		}
+		caSecret, err = c.client.CoreV1().Secrets(env.GetAntreaNamespace()).Create(context.TODO(), caSecret, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+		klog.Info("Created Secret for self-signed IPsec root CA")
+	}
+	caCertificate, err := certutil.ParseCertsPEM(caSecret.Data[corev1.TLSCertKey])
+	if err != nil {
+		return err
+	}
+	if len(caCertificate) == 0 {
+		return fmt.Errorf("CA certificate is empty")
+	}
+	privateKey, err := keyutil.ParsePrivateKeyPEM(caSecret.Data[corev1.TLSPrivateKeyKey])
+	if err != nil {
+		return err
+	}
+	priv, ok := privateKey.(crypto.Signer)
+	if !ok {
+		return fmt.Errorf("error reading CA: key did not implement crypto.Signer")
+	}
+	ca := &certificateAuthority{
+		RawCert:     caSecret.Data[corev1.TLSCertKey],
+		RawKey:      caSecret.Data[corev1.TLSPrivateKeyKey],
+		Certificate: caCertificate[0],
+		PrivateKey:  priv,
+	}
+	c.certificateAuthority.Store(ca)
+	desiredConfigMapData := map[string]string{
+		rootCACertKey: string(caSecret.Data[corev1.TLSCertKey]),
+	}
+	caConfigMap, err := c.configMapLister.ConfigMaps(env.GetAntreaNamespace()).Get(ipsecRootCAName)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		caConfigMap = &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      ipsecRootCAName,
+				Namespace: env.GetAntreaNamespace(),
+			},
+			Data: desiredConfigMapData,
+		}
+		caConfigMap, err = c.client.CoreV1().ConfigMaps(env.GetAntreaNamespace()).Create(context.TODO(), caConfigMap, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+		klog.InfoS("Created ConfigMap for self-signed IPsec root CA")
+	}
+	if !reflect.DeepEqual(desiredConfigMapData, caConfigMap.Data) {
+		toUpdate := caConfigMap.DeepCopy()
+		toUpdate.Data = desiredConfigMapData
+		_, err = c.client.CoreV1().ConfigMaps(env.GetAntreaNamespace()).Update(context.TODO(), toUpdate, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *IPsecCSRSigningController) csrWorker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+// watchSecretChanges use watch API directly to watch for Secret changes.
+// Antrea Controller should not have List permission for Secrets.
+func (c *IPsecCSRSigningController) watchSecretChanges(endCh <-chan struct{}) error {
+	// re-queue in case of hitting any errors when creating a new watcher.
+	c.fixturesQueue.Add(workerItemKey)
```

Comment:
It's likely useless to trigger a resync if watching the secret fails, why this is needed?

Reply:
I think it will fall back to check periodically if creating Secret watcher fails. But I am ok to remove it.

Reply:
I think it will fall back to periodical check anyway with or without this line? This line only affects whether to do a secret sync when it failes to watch the secret. But if watch API fails, I assume get secret would fail too?

Reply:
It's a really comprehensive consideration, but I think we shouldn't add code for such misconfiguration case, otherwise too many controversial code can be added.

Reply:
ok. removed.